napari/layers/labels/labels.py

```diff
@@ -893,6 +894,31 @@ def get_color(self, label):
             col = self.colormap.map(val)[0]
         return col
 
+    def _get_value_ray(
+        self, start_point, end_point, dims_displayed: List[int]
+    ):
+        """get the first non-background value encountered along a ray"""
+        if len(dims_displayed) == 3:
+            # only use get_value_ray on 3D for now
+            start_point = start_point[dims_displayed]
+            end_point = end_point[dims_displayed]
+            sample_ray = end_point - start_point
+            length_sample_vector = np.linalg.norm(sample_ray)
+            n_points = int(2 * length_sample_vector)
+            sample_points = np.linspace(
+                start_point, end_point, n_points, endpoint=True
+            ).astype(int)
+            im_slice = self._slice.image.raw
+            for point in sample_points:
+                point = clamp_point_to_bounding_box(
+                    point, self._display_bounding_box(dims_displayed)
+                ).astype(int)
+                value = im_slice[point[0], point[1], point[2]]
+                if value != 0:
+                    return value
```

Comment:
@kevinyamauchi did you consciously do a for-loop here vs NumPy? It's hard to tell which will be faster here, because NumPy won't let you do early termination, but this would have been my instinct:

```python
# btw: possibly only do `astype(int)` once,
# rather than both after linspace and after clamping?
clamped = clamp_points_to_bounding_box(
    sample_points, self._display_bounding_box(dims_displayed)
).astype(int)
values = im_slice[tuple(clamped.T)]
nz = values[np.flatnonzero(values)[0]]
```

Still, given napari's current data size limitations, this is probably O(500) values, so shouldn't be too dramatic either way... But might be worth keeping an eye on this.

Reply:
My rationale for the for loop was to allow for early termination. However, I didn't do any profiling when I made that decision 😬 . I went ahead and did some quick and dirty timing (below) and the tl;dr is that the numpy approach is faster (~30), but it's pretty fast either way for the size of our data. That being said, it's a quick change, so we might as well use the faster (numpy) approach. I think it's good to be optimistic about the size of volumes we will be able to render 😉 

average time for finding value in a volume with shape (500, 500, 500):

- for loop with 250 iterations before termination: 134 µs
- numpy: 4 µs

<details>
  <summary>Profiling script</summary>
  
  ```python
   import numpy as np

nx = 500
ny = 500
nz = 500
image = np.random.random((nz, ny, nx))
sample_points = np.column_stack(
    (
        np.zeros((nz,), dtype=int),
        np.zeros((ny,), dtype=int),
        np.arange(nz).astype(int)
    )
)
mean_dist_to_value = len(sample_points) / 2


def loop_method():
    for i, point in enumerate(sample_points):
        value = image[point[0], point[1], point[2]]
        if i > mean_dist_to_value:
            break


def numpy_method():
    values = image[tuple(sample_points.T)]
    # nz = values[np.flatnonzero(values)[0]]


if __name__ == "__main__":
    import timeit
    n_tests = 100000
    print (timeit.timeit(loop_method, number=n_tests) / n_tests)
    print (timeit.timeit(numpy_method, number=n_tests) / n_tests)
  ```
</details>

Reply:
Done! Also added a couple of tests to make sure it's working for ndim with transpose/roll.